internal/services/mssql/mssql_database_resource.go

```diff
@@ -1004,6 +1012,10 @@ func resourceMsSqlDatabaseUpdate(d *pluginsdk.ResourceData, meta interface{}) er
 		props.RecoverableDatabaseId = pointer.To(d.Get("recover_database_id").(string))
 	}
 
+	if d.HasChange("recovery_point_id") {
+		props.RecoveryServicesRecoveryPointId = pointer.To(d.Get("recover_database_id").(string))
```

Comment:
should this be `recovery_point_id`?

```suggestion
		props.RecoveryServicesRecoveryPointId = pointer.To(d.Get("recovery_point_id").(string))
```